### ISSUE TITLE
mixing of stdout/stderr with --parallel

### ISSUE THREAD

**Issue by ltirrell** [common-workflow-language/cwltool#903]
## Expected Behavior
- the stdout of different steps ran in parallel when using `--parallel` shouldn't be mixxed
- it would be nice if `cwltool` had an option to output a log file

## Actual Behavior
When running a workflow with the `--parallel` flag, stdout/stderr from steps running in parallel get mixed together. Would it be possible to prevent this in some way? Maybe by waiting until the step is complete before logging? Another solution could be prepending the job name to each line of the output (so each line would start with `[job workflow_name] `) when a flag is used, like `--timestamps`. This could be useful even when `--parallel` isn't used.

It would also be useful if `cwltool` had a flag to output logging information to a file as well as stdout/stderr.

## Workflow Code
I don't have a simple workflow showing thishat I can share. I can create one if that is useful.

## Your Environment
* cwltool version: 1.0.20180912090223

**Comment by mr-c** [comment]
Hello @ltirrell , this is a good request and similar to one that @FarahZKhan and @stain made for CWLProv (`--provenance`)

A question for all three of you: how do you want the logs organized? Separate files for STDERR and STDOUT? A combined file? With timestamps?

**Comment by ltirrell** [comment]
I could go either way about separate files. It seems that cwltool's STDOUT is just the json with output information, so if this is either at the end of the log or a separate file that is fine.

I like the `--timestamps` option (though I'd probably always keep it on myself). It'd be nice to have options for two logging levels, depending on how verbose you want them to be:
  - basic info including step/workflow star and completion status, where output is cached, and the commands ran. Note that the command is divided over many lines, I'd prefer if each line was prepended with `[timestamp] [description] `, instead of what is currently done.
```
[2018-09-12 19:51:59] [workflow main] starting step make_output
[2018-09-12 19:51:59] [step make_output] start
[2018-09-12 19:51:59] [job make_output] Output of job will be cached in /tmp/subjects/cache/d12d4af643fc08c35841382b68d0cf6b
[2018-09-12 19:51:59] [job make_output] /tmp/path/name$ command \
    argument1 \
    argument2 ... \
[2018-09-12 19:52:05] [job make_output] completed success
[2018-09-12 19:52:05] [step make_output] completed success
````
- the complete terminal output of the commands that are ran, with each line prepended with `[timestamp] [description] `. This would be useful for sorting through the log later. Also, if log output between different parallel steps are mixed, then they at least have information to help identify where they come from later.

If these 2 levels are separate files, that may also be useful (but I'm happy with one). The first is useful for just checking the correct commands ran, and checking where the workflow is currently. The second one is useful for debugging.
